From 275a70e4560df820549f100a3cbc47fa28245910 Mon Sep 17 00:00:00 2001
Date: Wed, 22 Oct 2025 09:39:51 -0400
Subject: [PATCH 1/2] Issue #3552735: Fix CSS aggregation by moving dialog.css
 weight adjustment to hook_library_info_alter().

Problem/Motivation: When CSS aggregation is enabled, the dialog.css reordering implemented in hook_css_alter() caused missing or invalid aggregated CSS files.
---
 gin_toolbar.module | 26 +++++++-------------------
 1 file changed, 7 insertions(+), 19 deletions(-)

diff --git a/gin_toolbar.module b/gin_toolbar.module
index 0996a51..454d61f 100644
--- a/gin_toolbar.module
+++ b/gin_toolbar.module
@@ -207,6 +207,13 @@ function gin_toolbar_library_info_alter(&$libraries, $extension) {
   if ($extension == 'workbench' && isset($libraries['workbench.toolbar'])) {
     $libraries['workbench.toolbar']['dependencies'][] = 'gin/workbench';
   }
+
+  // Set weight on gin/dialog CSS to ensure it loads after the theme CSS (weight 99)
+  if ($extension == 'gin' && isset($libraries['dialog']['css']['theme'])) {
+    foreach ($libraries['dialog']['css']['theme'] as $file => &$options) {
+      $options['weight'] = 101;
+    }
+  }
 }

 /**
@@ -399,25 +406,6 @@ function gin_toolbar_ckeditor_css_alter(array &$css) {
   }
 }

-/**
- * Set Gin_toolbar CSS on top of all other CSS files.
- */
-function gin_toolbar_css_alter(&$css, $assets) {
-  // Are we relevant?
-  if (!_gin_toolbar_gin_is_active()) {
-    return;
-  }
-
-  $path = \Drupal::service('extension.list.theme')->getPath('gin') . '/dist/css/components/dialog.css';
-
-  if (isset($css[$path])) {
-    // Use anything greater than 100 to have it load after the theme
-    // as CSS_AGGREGATE_THEME is set to 100.
-    // Let's be on the safe side and assign a high number to it.
-    $css[$path]['group'] = 101;
-  }
-}
-
 /**
  * Implements hook_toolbar_alter().
  */
--
GitLab


From 9bf708a3ba025c1113ec9211dbbe75d234842c8c Mon Sep 17 00:00:00 2001
Date: Wed, 22 Oct 2025 09:59:50 -0400
Subject: [PATCH 2/2] Issue #3552735: Fixing PHPCS errors

---
 gin_toolbar.module | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gin_toolbar.module b/gin_toolbar.module
index 454d61f..4b35e9f 100644
--- a/gin_toolbar.module
+++ b/gin_toolbar.module
@@ -208,7 +208,7 @@ function gin_toolbar_library_info_alter(&$libraries, $extension) {
     $libraries['workbench.toolbar']['dependencies'][] = 'gin/workbench';
   }

-  // Set weight on gin/dialog CSS to ensure it loads after the theme CSS (weight 99)
+  // Set weight on gin/dialog CSS so it loads after theme CSS (weight 99).
   if ($extension == 'gin' && isset($libraries['dialog']['css']['theme'])) {
     foreach ($libraries['dialog']['css']['theme'] as $file => &$options) {
       $options['weight'] = 101;
--
GitLab
